File: packages/urdu-js/tests/normalization/hamza.test.ts
import { describe, it, expect } from 'vitest'
import { normalizeHamza } from '../../src/normalization/hamza.js'

describe('normalizeHamza', () => {
  it('empty', () => expect(normalizeHamza('')).toBe(''))
  it('أ (U+0623) → ا', () => expect(normalizeHamza('\u0623')).toBe('\u0627'))
  it('ؤ (U+0624) → و', () => expect(normalizeHamza('\u0624')).toBe('\u0648'))
  it('isolated hamza (U+0621) unchanged', () => expect(normalizeHamza('\u0621')).toBe('\u0621'))
  it('ئ (U+0626) unchanged', () => expect(normalizeHamza('\u0626')).toBe('\u0626'))
})

File: packages/urdu-dart/test/normalization/hamza_test.dart
import 'package:test/test.dart';
import 'package:urdu_tools/urdu_tools.dart';

void main() {
  group('normalizeHamza', () {
    test('empty', () => expect(normalizeHamza(''), equals('')));
    test('أ (U+0623) -> ا',
        () => expect(normalizeHamza('\u0623'), equals('\u0627')));
    test('ؤ  (U+0624) -> و',
        () => expect(normalizeHamza('\u0624'), equals('\u0648')));
    test('isolated hamza (U+0621) unchanged',
        () => expect(normalizeHamza('\u0621'), equals('\u0621')));
    test('ئ (U+0626) unchanged',
        () => expect(normalizeHamza('\u0626'), equals('\u0626')));
  });
}
